src/chunk.c

```diff
@@ -3278,13 +3190,44 @@ ts_chunk_do_drop_chunks(Hypertable *ht, int64 older_than, int64 newer_than, int3
 	DEBUG_WAITPOINT("drop_chunks_chunks_found");
 
 	if (has_continuous_aggs)
-		ts_chunk_drop_process_invalidations(ht->main_table_relid,
-											older_than,
-											newer_than,
-											chunks,
-											num_chunks);
+	{
+		int i;
 
-	for (; i < num_chunks; i++)
+		/* Exclusively lock all affected chunks and refresh+invalidate the
+		 * dropped regions. Locking prevents further modification of the
+		 * dropped region during this transaction, which allows moving the
+		 * invalidation threshold without having to worry about new
+		 * invalidations while refreshing. */
+		for (i = 0; i < num_chunks; i++)
+		{
+			int64 start = chunks[i].cube->slices[0]->fd.range_start;
+			int64 end = chunks[i].cube->slices[0]->fd.range_end;
+
+			LockRelationOid(chunks[i].table_id, ExclusiveLock);
```

Comment:
Will this lock freed automatically in case of an error?

Reply:
Yes, released at transaction end or abort.